pkg/agent/config.go

```diff
@@ -43,7 +43,7 @@ type Endpoint struct {
 	Path     string `yaml:"path"`
 }
 
-type dataGatherer struct {
+type DataGatherer struct {
 	Kind     string
 	Name     string
 	DataPath string
```

Comment:
If this field is now exported I recommend adding `yaml:""` tags
```suggestion
type DataGatherer struct {
	Kind     string `yaml:"kind"`
	Name     string `yaml:"name"`
	DataPath string `yaml:"data_path"`
```